packages/compass-sidebar/src/components/use-filtered-connections.spec.ts

```diff
@@ -626,6 +642,27 @@ describe('useFilteredConnections', function () {
       });
     });
 
+    it('should filter collection items and database items using dot notation', async function () {
+      const { result } = renderHookWithContext(useFilteredConnections, {
+        initialProps: {
+          connections: mockSidebarConnections,
+          filter: {
+            regex: new RegExp('ready_1_1.coll_ready_shared_name', 'i'), // this matches only coll_ready_shared_name collection in ready_1_1 database
```

Comment:
While this works, the `.` has special meaning and I believe it should be escaped here to capture the intent of the change.

```suggestion
            regex: new RegExp('ready_1_1\.coll_ready_shared_name', 'i'), // this matches only coll_ready_shared_name collection in ready_1_1 database
```

Reply:
Fixed!